java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -85,4 +107,85 @@ public void ping_with_message_with_route() {
         String data = clusterClient.ping("H3LL0", ALL_PRIMARIES).get();
         assertEquals("H3LL0", data);
     }
+
+    @Test
+    @SneakyThrows
+    public void info_without_options() {
+        ClusterValue<String> data = clusterClient.info().get();
+        assertTrue(data.hasMultiData());
+        for (String info : data.getMultiValue().values()) {
+            for (String section : DEFAULT_INFO_SECTIONS) {
+                assertTrue(info.contains("# " + section), "Section " + section + " is missing");
+            }
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void info_with_route() {
+        ClusterValue<String> data = clusterClient.info(RANDOM).get();
+        assertTrue(data.hasSingleData());
+        String infoData = data.getSingleValue();
+        for (String section : DEFAULT_INFO_SECTIONS) {
+            assertTrue(infoData.contains("# " + section), "Section " + section + " is missing");
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void info_with_multiple_options() {
+        InfoOptions.InfoOptionsBuilder builder = InfoOptions.builder().section(CLUSTER);
+        if (REDIS_VERSION.feature() >= 7) {
+            builder.section(CPU).section(MEMORY);
+        }
+        InfoOptions options = builder.build();
+        ClusterValue<String> data = clusterClient.info(options).get();
+        for (String info : data.getMultiValue().values()) {
+            for (String section :  options.toArgs()) {
+                assertTrue(info.toLowerCase().contains("# " + section.toLowerCase()), "Section " + section + " is missing");
+            }
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void info_with_everything_option() {
+        InfoOptions options = InfoOptions.builder().section(EVERYTHING).build();
+        ClusterValue<String> data = clusterClient.info(options).get();
+        assertTrue(data.hasMultiData());
+        for (String info : data.getMultiValue().values()) {
+            for (String section : EVERYTHING_INFO_SECTIONS) {
+                assertTrue(info.contains("# " + section), "Section " + section + " is missing");
+            }
+        }
+    }
+
+    @Test
+    @SneakyThrows
+    public void info_with_routing_and_options() {
+        ClusterValue<Object> slotData = clusterClient.customCommand(new String[] {"cluster", "slots"}).get();
+        /*
+        Nested Object arrays like
+        1) 1) (integer) 0
+           2) (integer) 5460
+           3) 1) "127.0.0.1"
+              2) (integer) 7000
+              3) "92d73b6eb847604b63c7f7cbbf39b148acdd1318"
+              4) (empty array)
+        */
+        // Extracting first slot key
```

Comment:
[0])[2])[2] is the node ID == "92d73b6eb847604b63c7f7cbbf39b148acdd1318", why do you need it? why do you need to run cluster slots at all?

If you want to have a routing of a primary you can simply do `new SlotKeyRoute("foo", PRIMARY);`, we're covering all slots when we create the cluster so it would simply send the request to the primary that holds the slot of crc16("foo")

Reply:
This is just testing that we _can_ do this.  It's maybe not the most efficient command.